packages/client/ui/react-ui/src/components/embed/v3/crypto/SignerConnectionHandler.tsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from "react";
+import type { CrossmintEmbeddedCheckoutV3Props, EmbeddedCheckoutV3IFrameEmitter } from "@crossmint/client-sdk-base";
+
+export function SignerConnectionHandler({
+    props,
+    iframeClient,
+}: { props: CrossmintEmbeddedCheckoutV3Props; iframeClient: EmbeddedCheckoutV3IFrameEmitter | null }) {
+    useEffect(() => {
+        if (iframeClient == null) {
+            return;
+        }
+
+        const signTransactionListener = iframeClient.on(
+            "crypto:send-transaction",
+            async ({ serializedTransaction }) => {
+                try {
+                    const tx = await props.payment.crypto.signer!.handleSignAndSendTransaction(serializedTransaction);
```

Comment:
dont use `!`

i dont think u need to pass in all the props, instead just pass in the signer then u dont need to use `!`